src/Files.App/Utils/Serialization/Implementation/DefaultSettingsSerializer.cs

```diff
@@ -14,7 +14,7 @@ internal sealed class DefaultSettingsSerializer : ISettingsSerializer
 
 		public bool CreateFile(string path)
 		{
-			CreateDirectoryFromApp(Path.GetDirectoryName(path), IntPtr.Zero);
+			PInvoke.CreateDirectoryFromApp(Path.GetDirectoryName(path), new SECURITY_ATTRIBUTES());
 
 			var hFile = CreateFileFromApp(path, GENERIC_READ, FILE_SHARE_READ, IntPtr.Zero, OPEN_ALWAYS, (uint)File_Attributes.BackupSemantics, IntPtr.Zero);
 			if (hFile.IsHandleInvalid())
```

Comment:
What does `SECURITY_ATTRIBUTES` represent?

Reply:
Shouldn't that be nint.Zero or something?

Also, in fileapifromapp.h there should be more params.

Can you send me a screenshot of the definition of the function?

Reply:
> What does `SECURITY_ATTRIBUTES` represent?

From the docs:
The **lpSecurityDescriptor** member of the structure specifies a security descriptor for the new directory. If *lpSecurityAttributes* is **NULL**, the directory gets a default security descriptor. The ACLs in the default security descriptor for a directory are inherited from its parent directory. The target file system must support security on files and directories for this parameter to have an effect.

Checking the generated source, if null will do the same thing inside the method.

![image](https://github.com/files-community/Files/assets/52143624/c1bde954-4e5e-4fba-9ec7-ee421699cb55)


We can just pass null or create the default value like I'm doing , what you think?

Reply:
Passing null, default or nint.Zero is valid instead.

Reply:
`nint.Zero` is not valid, passing null instead.

Reply:
Thanks!